pyccel/codegen/printing/pycode.py

```diff
@@ -283,10 +272,7 @@ def _print_IndexedElement(self, expr):
                 indices = indices[0]
 
             indices = [self._print(i) for i in indices]
-            if expr.pyccel_staging != 'syntactic' and isinstance(expr.base.class_type, (HomogeneousTupleType, HomogeneousListType)):
-                indices = ']['.join(i for i in indices)
-            else:
-                indices = ','.join(i for i in indices)
+            indices = ','.join(i for i in indices)
```

Comment:
The previous code here:
```python
            if expr.pyccel_staging != 'syntactic' and isinstance(expr.base.class_type, (HomogeneousTupleType, HomogeneousListType)):
                indices = ']['.join(i for i in indices)
```
was a hack. Before this PR the syntactic stage condenses things like `a[1][2][3]` into 1 `IndexedElement` `a[1,2,3]`. This makes sense if `a` is an ND array, but it doesn't otherwise. For tuples and lists as a workaround we were assuming that we never have a tuple/list of ND arrays. The hack was extended to lists recently fixing https://github.com/pyccel/pyccel/issues/1732. We decided it was a reasonable workaround given that we knew that this PR was coming soon (it was already open) and would fix it cleanly.

Reply:
OK, much better to live without that hack!

BTW, is there a reason for not writing just this?
``` python
 indices = ','.join(indices)
```

Reply:
No reason: 2e524617d